azurerm/resource_arm_nat_gateway_test.go

```diff
@@ -0,0 +1,201 @@
+		CheckDestroy: testCheckAzureRMNatGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNatGateway_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNatGatewayExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNatGateway_complete(t *testing.T) {
+	resourceName := "azurerm_nat_gateway.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNatGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNatGateway_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNatGatewayExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_address_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_address_ids.0", "${azurerm_public_ip.<%= resource_id_hint -%>.id}"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_prefix_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_prefix_ids.0", "${azurerm_public_ip_prefix.<%= resource_id_hint -%>.id}"),
+					resource.TestCheckResourceAttr(resourceName, "sku", "Standard"),
+					resource.TestCheckResourceAttr(resourceName, "idle_timeout_in_minutes", "10"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNatGateway_update(t *testing.T) {
+	resourceName := "azurerm_nat_gateway.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNatGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNatGateway_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNatGatewayExists(resourceName),
+				),
+			},
+			{
+				Config: testAccAzureRMNatGateway_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNatGatewayExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_address_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_address_ids.0", "${azurerm_public_ip.<%= resource_id_hint -%>.id}"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_prefix_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "public_ip_prefix_ids.0", "${azurerm_public_ip_prefix.<%= resource_id_hint -%>.id}"),
+					resource.TestCheckResourceAttr(resourceName, "sku", "Standard"),
+					resource.TestCheckResourceAttr(resourceName, "idle_timeout_in_minutes", "10"),
+				),
+			},
```

Comment:
Can we add another step here that reverts the resource back to the `basic` state?
```suggestion
			},
			{ 
				Config: testAccAzureRMNatGateway_basic(ri, location), 
				Check: resource.ComposeTestCheckFunc( 
					testCheckAzureRMNatGatewayExists(resourceName), 
				), 
			},
```

Reply:
This is unavailable currently due to this [issue](https://github.com/terraform-providers/terraform-provider-azurerm/issues/4450)